Helper/PaymentResponseHandler.php

```diff
@@ -247,11 +250,19 @@ public function handlePaymentResponse($paymentsResponse, $payment, $order = null
                         }
                     } else {
                         $order = $payment->getOrder();
-                        $this->recurringHelper->createAdyenBillingAgreement(
-                            $order,
-                            $paymentsResponse['additionalData'],
-                            $payment->getAdditionalInformation()
-                        );
+                        $recurringMode = $this->configHelper->getCardRecurringMode($storeId);
+
+                        // if adyen tokenization set up, create entry in paypal_billing_agreement table
+                        if ($recurringMode === self::ADYEN_TOKENIZATION) {
+                            $this->recurringHelper->createAdyenBillingAgreement(
+                                $order,
+                                $paymentsResponse['additionalData'],
+                                $payment->getAdditionalInformation()
+                            );
+                            // if vault set up, create entry in vault_payment_token table
+                        } elseif ($recurringMode === self::VAULT && $paymentInstanceCode === AdyenCcConfigProvider::CODE) {
```

Comment:
If we add `$paymentInstanceCode === AdyenCcConfigProvider::CODE` condition above then this one will be redundant.